specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2022-06-01-preview/ConfidentialWatchlists.json

```diff
@@ -0,0 +1,441 @@
+              "$ref": "../../../common/2.0/types.json#/definitions/CloudError"
+            }
+          }
+        }
+      },
+      "delete": {
+        "x-ms-examples": {
+          "Delete a confidential watchlist.": {
+            "$ref": "./examples/confidentialWatchlists/DeleteConfidentialWatchlist.json"
+          }
+        },
+        "tags": [
+          "ConfidentialWatchlists"
+        ],
+        "description": "Delete a confidential watchlist.",
+        "operationId": "ConfidentialWatchlists_Delete",
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "../../../common/2.0/types.json#/parameters/WorkspaceName"
+          },
+          {
+            "$ref": "./common/WatchlistTypes.json#/parameters/WatchlistAlias"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "headers": {
+              "Azure-AsyncOperation": {
+                "description": "Contains the status URL on which clients are expected to poll the status of the delete operation.",
+                "type": "string"
+              }
+            }
+          },
+          "204": {
+            "description": "No Content"
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "../../../common/2.0/types.json#/definitions/CloudError"
+            }
+          }
+        }
+      },
+      "put": {
+        "x-ms-examples": {
+          "Creates or updates a confidential watchlist.": {
+            "$ref": "./examples/confidentialWatchlists/CreateConfidentialWatchlist.json"
+          },
+          "Creates or updates a confidential watchlist and bulk creates watchlist items.": {
+            "$ref": "./examples/confidentialWatchlists/CreateConfidentialWatchlistAndWatchlistItems.json"
+          }
+        },
+        "tags": [
+          "ConfidentialWatchlists"
+        ],
+        "description": "Create or update a confidential Watchlist and its Watchlist Items (bulk creation, e.g. through text/csv content type). To create a Watchlist and its Items, we should call this endpoint with either rawContent or a valid SAR URI and contentType properties. The rawContent is mainly used for small watchlist (content size below 3.8 MB). The SAS URI enables the creation of large watchlist, where the content size can go up to 500 MB. The status of processing such large file can be polled through the URL returned in Azure-AsyncOperation header.",
```

Comment:
>To create a Watchlist and its Items, we should call this endpoint with either rawContent or a valid SAR URI and contentType properties. The rawContent is mainly used for small watchlist (content size below 3.8 MB). The SAS URI enables the creation of large watchlist, where the content size can go up to 500 MB. The status of processing such large file can be polled through the URL returned in Azure-AsyncOperation header.", [](http://example.com/codeflow?start=143&length=425)

Can you elaborate the scenarios regarding these APIs? I wonder if data-plane api is a better fit for watchlist items given the size and the implicit resource creation.

Reply:
A watchlist can have up to 10 million records. ARM payload has an upper limit of 4MB and does not fit with customers needs. Since 2022-01-01-preview, we support large file through SASUri : customers upload a large file (up to 500 MB) to their own Azure Storage, then generate a SASUri and use it in the PUT. API processes the file from the SASUri.  We had a discussion with ARM before designing the large watchlist. I am curious if you can share a link about data plan.

Reply:
I mean since you are bypassing control-plain by creating the child resources implictly, not sure why it needs to go through ARM. What are you gaining from ARM? I wonder your RP providing a data plan API to upload watchlist items in bulk can be an option.

Reply:
We will consider it in future api-version.